src/Catel.MVVM/Catel.MVVM.Shared/Services/UIVisualizerService.wpf.cs

```diff
@@ -126,43 +126,36 @@ protected virtual void HandleCloseSubscription(object window, object data, Event
         /// <param name="window">The window.</param>
         /// <param name="showModal">If <c>true</c>, the window should be shown as modal.</param>
         /// <returns><c>true</c> if the window is closed with success; otherwise <c>false</c> or <c>null</c>.</returns>
-        protected virtual bool? ShowWindow(FrameworkElement window, bool showModal)
+        protected virtual Task<bool?> ShowSystemWindowAsync(FrameworkElement window, bool showModal)
         {
-            if (showModal)
-            {
-                var showDialogMethodInfo = window.GetType().GetMethodEx("ShowDialog");
-                if (showDialogMethodInfo != null)
-                {
-                    // Child window does not have a ShowDialog, so not null is allowed
-                    bool? result = null;
+            Argument.InheritsFrom<System.Windows.Window>("window", window);
 
-                    window.Dispatcher.InvokeIfRequired(() =>
+            return Task.Run(() => {
```

Comment:
No need to use Task.Run. Use TaskCompletionSource instead which gets returned as a task.